betty/job/executor.py

```diff
@@ -0,0 +1,99 @@
+"""
+Job execution solutions.
+"""
+
+from __future__ import annotations
+
+from asyncio import CancelledError, Task, create_task, gather, sleep
+from contextlib import suppress
+from typing import TYPE_CHECKING, Self, TypeVar, final
+
+from betty.job import Context
+from betty.job.scheduler import Closed, Scheduler
+
+if TYPE_CHECKING:
+    from types import TracebackType
+
+
+_ContextCoT = TypeVar("_ContextCoT", bound=Context, covariant=True)
+
+
+@final
+class Executor:
+    """
```

Comment:
Extract an interface and rename this implementation to `AsyncExecutor` in an `asyncio` submodule